### DISCUSSION TITLE
How do you get a VTimeZone from DtStart in ical4j 4.x?

### DISCUSSION THREAD

**Discussion by mosesn** [ical4j/ical4j#752] [Q&A] [answered]
We're migrating from 3.x to 4.x, and we used to be able to do:

`DtStart#getTimeZone` (returning a `TimeZone`) and then `TimeZone#getVTimeZone`. However, the `DtStart#getTimeZone` API was removed. What's the new way of doing this?

**Comment by benfortuna** [comment, marked as answer]
To get a VTIMEZONE from a date property with iCal4j 4.x you first need to get the `ZoneId` and use it to look up the timezone registry.

With iCal4j 3.x this lookup was encapsulated in `DtStart#getTimeZone`, but has been decoupled to support the new date/time API.

### To get the zone id use:

```java
TzId tzid = prop.getRequiredParameter("TZID");
ZoneId zoneid = tzid.toZoneId(); // or;

TimeZoneRegistry tzRegistry = TimeZoneRegistryFactory.getInstance().createRegistry(); // use this registry in calendar builder, etc. to register tz definitions from input data
ZoneId zoneid = tzid.toZoneId(tzRegistry);
```

### To get the VTIMEZONE use:

```java
TimeZone tz = tzRegistry.getTimeZone(zoneid.getId());
VTimeZone vtz = tz.getVTimeZone();
```

I haven't fully tested this code but should be something like that..
